app/features/lfg/lfg-types.ts

```diff
@@ -45,3 +45,109 @@ type MinTierFilter = {
   _tag: "MinTier";
   tier: TierName;
 };
+
+const typeToNum: Map<LFGType, number> = new Map([
+  ["PLAYER_FOR_TEAM", 0],
+  ["TEAM_FOR_PLAYER", 1],
+  ["TEAM_FOR_COACH", 2],
+  ["COACH_FOR_TEAM", 3],
+]);
+
+const numToType = new Map(
+  Array.from(typeToNum).map(([type, num]) => [`${num}`, type]),
+);
+
+const tierToNum = new Map(
+  TIERS.map((tier, index) => {
+    return [tier.name, `${index}`];
+  }),
+);
+
+const numToTier = new Map(
+  Array.from(tierToNum).map(([tier, num]) => [`${num}`, tier]),
+);
+
+export function filterToSmallStr(filter: LFGFilter): string {
+  switch (filter._tag) {
+    case "Weapon": {
+      const weapons = filter.weaponSplIds.map((wid) => `${wid}`).join(",");
+      return `w.${weapons}`;
+    }
+    case "Type":
+      return `t.${typeToNum.get(filter.type)}`;
+    case "Timezone":
+      return `tz.${filter.maxHourDifference}`;
+    case "Language":
+      return `l.${filter.language}`;
+    case "PlusTier":
+      return `pt.${filter.tier}`;
+    case "MaxTier":
+      return `mx.${tierToNum.get(filter.tier)}`;
+    case "MinTier":
+      return `mn.${tierToNum.get(filter.tier)}`;
+  }
```

Comment:
We can use `assertUnreachable` in the default case here to ensure at a build time all the cases have been handled

Reply:
I found not having a return outside the switch statement does the same thing implicitly, but I like the explicitness of your option.